app/modules/wallet/addresses/addresses.controller.ts

```diff
@@ -0,0 +1,56 @@
+import * as restify from 'restify';
+import { InternalServerError, NotFoundError } from 'restify-errors';
+import IController from '../../../interfaces/utils/IController';
+import IRequest from '../../../interfaces/utils/IServer';
+import WalletAddress from '../../../models/wallet_address.model';
+import { IWallet } from '../../../interfaces/models';
+
+export default class WalletsController implements IController {
+  public async put(req: IRequest, res: restify.Response, next: restify.Next) {
+    try {
+      const walletAddress = await WalletAddress.create({
+        chain: req.body.chain,
+        address: req.body.address,
+        name: req.body.name,
+        wallet_id: req.params.wallet
+      });
+
+      return res.send({
+        walletAddress
+      });
+    } catch (e) {
+      console.log(e);
+      throw new InternalServerError(e);
+    }
+    return next();
+  }
+
+  public async get(req: IRequest, res: restify.Response, next: restify.Next) {
```

Comment:
`getAll` should be used to retrieve all resources. `get` is used to find one.

Reply:
Retrieving single record here.

Reply:
You're using `findAll` thats why i said that, if retrieving single record, use `findOne`